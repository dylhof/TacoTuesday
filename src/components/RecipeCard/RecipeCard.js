import React from 'react';
import uuid from 'uuid/v4'

export const RecipeCard = ({ tacoRecipe }) => {
  const splitRecipe = tacoRecipe.recipe.split('\n');
  const jsxRecipeItems = splitRecipe.map(step => {
    if (!step.includes('=')) {
      return (
        <li key={uuid()}>
          {step}
        </li>
      )
    }
  });
  return (
    <div className='recipeCard'>
      <h3>{tacoRecipe.name}</h3>
      <ul dense={true} className='recipeList'>
        {jsxRecipeItems}
      </ul>
    </div>
  );
};

export default RecipeCard;