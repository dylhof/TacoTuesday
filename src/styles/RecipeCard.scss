.recipeCard {
  margin: 10px;
  padding: 10px;
  opacity: .8;
  min-width: 300px;
  border-radius: 5px;
  background-color: $highlight;

  .recipeListDiv{
  }
  
  .recipeList {
    margin: 10px;
    overflow: scroll;
    border: 2px solid $secondary;
    border-radius: 5px;
    padding: 3px;
  }
}

.homeRecipes {
  height: 300px;
  
  .homeList {
    height: 250px
  }
}