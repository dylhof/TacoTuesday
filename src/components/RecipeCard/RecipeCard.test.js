import React from 'react';
import { shallow } from 'enzyme';
import { RecipeCard } from './RecipeCard';

describe('RecipeCard', () => {
  let wrapper;
  let mockRecipe = {
    slug: 'Yum_Taco', 
    name: 'yummyTacos',
    recipe: 'lots of steps \n another step'}
  let mockClasses = {card: {}, list: {}}

  it('should match the snapshot', () => {
    //setup
    wrapper = shallow(
      <RecipeCard 
        tacoRecipe={mockRecipe}
        classes={mockClasses}/>)
    //expectation
    expect(wrapper).toMatchSnapshot()
  });
});