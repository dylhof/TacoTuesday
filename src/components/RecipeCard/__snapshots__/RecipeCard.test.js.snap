// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeCard should match the snapshot 1`] = `
<WithStyles(Card)
  className={Object {}}
>
  <h3>
    yummyTacos
  </h3>
  <WithStyles(List)
    className={Object {}}
    dense={true}
  >
    <WithStyles(ListItem)
      key="c088dedd-3834-4bef-ac76-219016891e46"
    >
      <WithStyles(ListItemText)>
        lots of steps 
      </WithStyles(ListItemText)>
    </WithStyles(ListItem)>
    <WithStyles(ListItem)
      key="3a3377c8-dfe8-4713-97bb-e04f4e70e970"
    >
      <WithStyles(ListItemText)>
         another step
      </WithStyles(ListItemText)>
    </WithStyles(ListItem)>
  </WithStyles(List)>
</WithStyles(Card)>
`;
